core/domain/improvements_services.py

```diff
@@ -0,0 +1,129 @@
+# coding: utf-8
+#
+# Copyright 2020 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Service functions related to Oppia improvement tasks."""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+import collections
+import itertools
+import operator
+
+from core.domain import improvements_domain
+from core.platform import models
+import feconf
+import python_utils
+
+(improvements_models,) = (
+    models.Registry.import_models([models.NAMES.improvements]))
+
+_MODEL = improvements_models.TaskEntryModel
```

Comment:
Not too keen on this -- typically service files make use of more than one storage model. Either inline it fully or use an alias that also includes the words "task" and "entry".